mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.java

```diff
@@ -251,6 +274,102 @@ public void onPause() {
         }
     }
 
+    @SuppressLint("StaticFieldLeak")
+    private void createShortcut(LinkedPage linkedPage) {
+        String iconFormat = PreferenceManager.getDefaultSharedPreferences(getContext())
+                .getString(Constants.PREFERENCE_ICON_FORMAT, "png");
+        new AsyncTask<Void, Void, IconCompat>() {
+
+            @Override
+            protected IconCompat doInBackground(Void... voids) {
+                Context context = getContext();
+                String url = null;
+                if (!TextUtils.isEmpty(linkedPage.iconPath())) {
+                    url = new Uri.Builder()
+                            .appendEncodedPath(linkedPage.iconPath())
+                            .appendQueryParameter("format", iconFormat)
+                            .toString();
+                }
+                IconCompat icon = null;
+                Connection connection = null;
+                try {
+                    connection = ConnectionFactory.getUsableConnection();
+                } catch (ConnectionException e) {
+                    // ignored
+                }
+
+                if (context == null || connection == null) {
+                    return null;
+                }
+
+                if (url != null) {
+                    /** Icon size is defined in {@link AdaptiveIconDrawable}. **/
+                    int foregroundSize = (int) Util.convertDpToPixel(46, context);
```

Comment:
I don't seem to be able to find those magic numbers (46/31) in `AdaptiveIconDrawable`'s sources...instead, it says this:
```
 * <li>Both foreground and background layers should be sized at 108 x 108 dp.</li>
 * <li>The inner 72 x 72 dp  of the icon appears within the masked viewport.</li>
 * <li>The outer 18 dp on each of the 4 sides of the layers is reserved for use by the system UI
 * surfaces to create interesting visual effects, such as parallax or pulsing.</li>
 * </ul>
```
So foregroundSize should be 72dp and border size should be 18dp ... what am I missing?

Reply:
First I tried it with these values, but then realized, that "masked viewport" and foreground size are two different things: If I set foregroundSize to 72dp and have a square homescreen icon, the icon from openHAB touches the border of the homescreen icon. So I made the foregroundSize smaller and the border bigger: 46 + (2 x 31) = 108 (the border is added on each site of the icon).
I should add that to the comment.